create_binaries.sh

```diff
@@ -16,36 +16,51 @@ for i in $TARGETARCH; do
 	[ "$dirarch" = "aarch64" ] && dirarch=arm64
 	[ "$dirarch" = "ppc64el" ] && dirarch=ppc64le
 	[ "$dirarch" = "powerpc64le" ] && dirarch=ppc64le
+
+	# where we place our output binaries
 	distarch=$DIST/$dirarch
 	mkdir -p $distarch
-	# if target arch is our arch, then no --host=<triple>
-
+	distarch=$(readlink -f $distarch)
+	# by making our obj/ dir be distinct per arch, we avoid potentially trouncing with .o files of wrong arch
+	objarch=$OBJ/$dirarch
+	mkdir -p $objarch
+	objarch=$(readlink -f $objarch)
 
+	# initial pwd
+	initpwd=$(pwd)
 
+	(
+	cd $objarch
+	# if target arch is our arch, then no --host=<triple>
 	HOSTARCH=
 	GCC=gcc
 	# are we cross-compiling
 	if [ "$i" != "$ARCH" ]; then
 		HOSTARCH="$i-linux-gnu"
 		GCC="$HOSTARCH-gcc"
 	fi
-	./configure  --with-protocols="bgp pipe static" --enable-ipv6=yes --enable-client=yes --enable-pthreads=yes --with-sysconfig=linux-v6 --build=$ARCH --host=$HOSTARCH
+	$initpwd/configure  --with-protocols="bgp pipe static" --enable-ipv6=yes --enable-client=yes --enable-pthreads=yes --with-sysconfig=linux-v6 --build=$ARCH --host=$HOSTARCH
 	make
 	# Remove the dynamic binaries and rerun make to create static binaries and store off the results
 	rm bird birdcl
 
 	# we need to force static compilation
-	make CC="$GCC --static"
+	# because of the circular dependency when bird runs in its own directory (birdc -> ./birdc), it will try to rebuild birdc, which fails when static
+	#    thus, always build explicit targets when doing --static
```

Comment:
Code change is fine, but I don't understand what's changed here.  (I.e. I don't understand the comment that you've added.)

Reply:
When you run `configure` in a dir other than the source directory, it sets `execdir=.`. This in turn changes the dependency in the generated `Makefile` from:

```makefile
bird: ../bird
```

to 

```makefile
bird: ./bird
```

Same for `birdcl`, `birdc`. 

This becomes a circular dependency, which causes it to always rebuild the 3 binaries, even if they exist.

In principle, this isn't a problem, except that `birdc` cannot be built with `gcc --static`. So just running:

```
make CC="$GCC static"
``` 

will fail, because it will try to rebuild `birdc` and fail. On the other hand, explicitly making the desired targets as follows:

```
make CC="$GCC --static" bird birdcl
```

ensures that just those are rebuilt (which is what we are trying to do anyways, see the previous line with `rm bird birdcl`), and everything works.

Yeah, this really should be handled better by `autoconf`, but I wasn't about to rejigger all of that, since it would take a much longer time to figure out, and it would be a change from the original `bird`, which we are trying to avoid by using wrappers.

Does that explain it?

Reply:
OK, thanks @deitch .  I still don't follow every detail here, but I get the picture of the twisty autoconf maze...

Reply:
> twisty autoconf maze

Yeah, that is a pretty accurate description. Took me too long to figure out why the build was failing when it should have been a simple change, but then I got it.